platform/android/java/lib/src/org/godotengine/godot/plugin/GodotPlugin.java

```diff
@@ -189,6 +190,22 @@ public void onGLSurfaceChanged(GL10 gl, int width, int height) {}
 	 */
 	public void onGLSurfaceCreated(GL10 gl, EGLConfig config) {}
 
+	/**
+	 * Invoked once per frame on the Vulkan thread after the frame is drawn.
+	 */
+	public void onVkDrawFrame() {}
+
+	/**
+	 * Called on the Vulkan thread after the surface is created and whenever the surface size
+	 * changes.
+	 */
+	public void onVkSurfaceChanged(Surface surface, int width, int height) {}
+
+	/**
+	 * Called on the Vulkan thread when the surface is created or recreated.
+	 */
+	public void onVkSurfaceCreated(Surface surface) {}
+
```

Comment:
I'd merge these with the `GL*` callbacks by renaming them to make them generic. For example, `onRenderDrawFrame`, `onRenderSurfaceChanged`, `onRenderSurfaceCreated`.

Reply:
Do you mean that the common functions would take both a `Surface` and a `GL10` as arguments, and one of them would be null depending on the renderer used?

Or maybe we could simply pass a `SurfaceView` argument instead, but I don't have an example of these callbacks being used so I'm not sure what the actual use cases are.